File: JavaScript/7js 3/solution_str_end.js
/*
|--------------------------------------------------------------------------
| String ends with?
|--------------------------------------------------------------------------
|
*/

function solution(str, ending) {
   const revEnd = ending.split("").reverse();
   const len = revEnd.length;
   const revStr = str.split("").reverse().splice(0, len);
   return revStr.join("") == revEnd.join("") ? true : false;
}


function solution(str, ending) {
   return str.endsWith(ending);
}


function solution(str, ending) {
   return ending === str.substr(str.length - ending.length, str.length);
}


function solution(str, ending) {
   const l = ending.length;
   const str = str.slice(-l);
   return str.match(ending) ? true : false;
}


console.log(solution('abcde', 'cde'), true)
console.log(solution('abcde', 'abc'), false)

File: Python/7py 3/solution_str_end.py
"""
|--------------------------------------------------------------------------
| String ends with?
|--------------------------------------------------------------------------
|
"""

def solution(string, ending):
   return True if ending == '' else string[len(string) - len(ending):] == ending


def solution(string, ending):
   return string.endswith(ending)


def solution(string, ending):
   return ending in string[-len(ending):]


print(solution('abcde', 'cde'), True)
print(solution('abcde', 'abc'), False)
print(solution('abcde', ''), True)